client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -84,6 +72,63 @@ class EditorMediaModalDetailItem extends Component {
 		);
 	}
 
+	handleOnRestoreClick = () => {
+		const { site, item, onRestore } = this.props;
+		onRestore( site && site.ID, item );
+	};
+
+	renderRestoreButton() {
+		const {
+			item,
+			translate
+		} = this.props;
+
+		//do a simple guid vs url check
+		const guidParts = url.parse( item.guid );
+		const URLParts = url.parse( item.URL );
+
+		if ( guidParts.pathname === URLParts.pathname ) {
+			return false;
+		}
+
+		return (
+			<Button
+				className={ classNames(
+					'editor-media-modal-detail__restore',
+				) }
+				onClick={ this.handleOnRestoreClick }
+				disabled={ isItemBeingUploaded( item ) }
+			>
+				<Gridicon
+					icon="refresh"
+					size={ 36 } />
+					{ translate( 'Restore Original' ) }
+			</Button>
+		);
+	}
+
+	renderImageEditorButtons( item, classname = 'is-desktop' ) {
+		const { site } = this.props;
+
+		if (
+			site.is_private ||
```

Comment:
Maybe we could add a comment clarifying the reasoning behind not enabling image editing on private sites here.